pkg/apis/monitoring/v1/types.go

```diff
@@ -483,25 +487,29 @@ type QueueConfig struct {
 // RemoteReadSpec defines the remote_read configuration for prometheus.
 // +k8s:openapi-gen=true
 type RemoteReadSpec struct {
-	//The URL of the endpoint to send samples to.
+	// The URL of the endpoint to send samples to.
 	URL string `json:"url"`
-	//An optional list of equality matchers which have to be present
+	// The name of the remote read queue, must be unique if specified. The name
+	// is used in metrics and logging in order to differentiate read
+	// configurations.  Only valid in Prometheus versions 2.15.0 and newer.
+	Name string `json:"name,omitempty"`
```

Comment:
this still needs wiring up to the config generation

Reply:
Do'h, added the wiring for read and write cases. Do you think this is worth a test as well? I am not seeing any other remote specific tests.

Reply:
I'm happy with this as is. Thanks!